streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java

```diff
@@ -2330,4 +2331,484 @@ public void shouldNotAllowNullNamedOnProcessWithStores() {
+        final Consumed<String, String> consumed = Consumed.with(Serdes.String(), Serdes.String());
+        final StreamsBuilder builder = new StreamsBuilder();
+        final String input = "input";
+        final String output = "output";
+
+        builder
+            .stream(input, consumed)
+            .toTable()
+            .groupBy(MockMapper.selectValueKeyValueMapper(), Grouped.with(Serdes.String(), Serdes.String()))
+            .count(Materialized.as("count"))
+            .toStream()
+            .to(output);
+
+        testCountHelper(builder, input, output);
+    }
+
+    private static void testCountHelper(final StreamsBuilder builder,
+                                        final String input,
+                                        final String output) {
+        try (
+            final TopologyTestDriver driver = new TopologyTestDriver(
+                builder.build(),
+                mkProperties(mkMap(
+                    mkEntry(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "dummy"),
+                    mkEntry(StreamsConfig.APPLICATION_ID_CONFIG, "test"),
+                    mkEntry(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory("kafka-test").getAbsolutePath())
+                )),
+                Instant.ofEpochMilli(0L))) {
+            final TestInputTopic<String, String> inputTopic =
+                driver.createInputTopic(input, new StringSerializer(), new StringSerializer(), Instant.ofEpochMilli(0L), Duration.ZERO);
+            final TestOutputTopic<String, Long> outputTopic =
+                driver.createOutputTopic(output, Serdes.String().deserializer(), Serdes.Long().deserializer());
+
+
+            inputTopic.pipeInput("A", "green", 10L);
+            inputTopic.pipeInput("B", "green", 9L);
+            inputTopic.pipeInput("A", "blue", 12L);
+            inputTopic.pipeInput("C", "yellow", 15L);
+            inputTopic.pipeInput("D", "green", 11L);
+
+            assertEquals(
+                asList(
+                    new TestRecord<>("green", 1L, Instant.ofEpochMilli(10)),
+                    new TestRecord<>("green", 2L, Instant.ofEpochMilli(10)),
+                    new TestRecord<>("green", 1L, Instant.ofEpochMilli(12)),
+                    new TestRecord<>("blue", 1L, Instant.ofEpochMilli(12)),
+                    new TestRecord<>("yellow", 1L, Instant.ofEpochMilli(15)),
+                    new TestRecord<>("green", 2L, Instant.ofEpochMilli(12))),
+                outputTopic.readRecordsToList());
+        }
+    }
+
+    @Test
+    public void shouldSupportMapValueTriggerMaterializedWithKTableFromKTStream() {
+        final Consumed<String, String> consumed = Consumed.with(Serdes.String(), Serdes.String());
+        final String input = "input";
+        final String output = "output";
+        final String storeName = "store";
+
+        builder.stream(input, consumed)
+            .toTable()
+            .mapValues(
+                value -> value.charAt(0) + 48,
```

Comment:
Maybe do `value -> value.charAt(0) - (int)'a',` to map `a` to zero, `b` to one etc (makes it easier to read the test IMHO -- compare my comment below)